### PR TITLE
feat(CD): CHK-25-40 Add pipeline for prod

### PR DESCRIPTION
**Pull request by infantesimone**
#### List of Changes

- add Prod pipeline 🚀

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .devops/deploy-pipelines.yml [hunk 1]
```diff
@@ -125,18 +132,18 @@ stages:
...
+    #   succeeded(),
+    #   eq(${{parameters.UAT_PROD_DEPLOY}}, true),
+    #   or(
+    #   eq(variables['Build.SourceBranch'], 'refs/heads/main'),
+    #   and(
+    #   startsWith(variables['Build.SourceBranch'], 'refs/tags'),
+    #   endsWith(variables['Build.SourceBranch'], '-RELEASE')
+    #   )
+    #   )
+    #   )
```
[the remark this answers is not in the document]
Comment (infantesimone) [reply .devops/deploy-pipelines.yml]: fixed :-)

File: .devops/deploy-pipelines.yml [hunk 2]
```diff
@@ -297,4 +304,255 @@ stages:
...
+                    GREEN_VERSION: $(green_version)
+
+  # --- START Deploy PROD --- #
+  - stage: "Prod_WaitForApproval"
+    displayName: 'PROD approval deployment'
+    dependsOn: [ Deploy_UAT_Green,Release ]
+    condition: |
+      and(
+        eq(${{parameters.UAT_PROD_DEPLOY}}, true),
+        in(dependencies.Deploy_UAT_Green.result, 'Succeeded'),
```
Comment (pietro-tota) [inline .devops/deploy-pipelines.yml:316]:
```suggestion
        eq(dependencies.Deploy_UAT_Green.result, 'Succeeded'),
```

File: .devops/deploy-pipelines.yml [hunk 3]
```diff
@@ -297,4 +304,255 @@ stages:
...
+              onTimeout: 'reject'
+
+  - stage: "Build_PROD_Blue"
+    displayName: 'PROD blue Build'
+    dependsOn: [ Prod_WaitForApproval,Release ]
+    condition: |
+      and(
+        eq(${{parameters.UAT_PROD_DEPLOY}}, true),
+        eq(${{parameters.PROD_SKIP_BLUE_DEPLOYMENT}}, False),
+        in(dependencies.Prod_WaitForApproval.result, 'Succeeded'),
```
Comment (pietro-tota) [inline .devops/deploy-pipelines.yml:341]:
```suggestion
        eq(dependencies.Prod_WaitForApproval.result, 'Succeeded'),
```

File: .devops/deploy-pipelines.yml [hunk 4]
```diff
@@ -297,4 +304,255 @@ stages:
...
+              targetType: "inline"
+              script: |
+                docker push $(PROD_CONTAINER_NAMESPACE)/$(K8S_IMAGE_REPOSITORY_NAME):$(Build.SourceVersion)
+  - stage: "Deploy_PROD_Blue"
+    displayName: 'PROD blue deployment'
+    condition: |
+      and(
+        eq(${{parameters.UAT_PROD_DEPLOY}}, true),
+        eq(${{parameters.PROD_SKIP_BLUE_DEPLOYMENT}}, False),
+        in(dependencies.Build_PROD_Blue.result, 'Succeeded'),
```
Comment (pietro-tota) [inline .devops/deploy-pipelines.yml:379]:
```suggestion
        eq(dependencies.Build_PROD_Blue.result, 'Succeeded'),
```

File: .devops/deploy-pipelines.yml [hunk 5]
```diff
@@ -297,4 +304,255 @@ stages:
...
+                    BLUE_VERSION: $(Build.SourceVersion)
+
+  - stage: "PROD_Green_WaitForApproval"
+    displayName: 'PROD green approval deployment'
+    ${{ if eq(parameters['PROD_SKIP_BLUE_DEPLOYMENT'], True) }}:
+      dependsOn: Prod_WaitForApproval
+      condition: |
+        and(
+          eq(${{parameters.UAT_PROD_DEPLOY}}, true),
+          in(dependencies.Prod_WaitForApproval.result, 'Succeeded')
```
Comment (pietro-tota) [inline .devops/deploy-pipelines.yml:423]:
```suggestion
          eq(dependencies.Prod_WaitForApproval.result, 'Succeeded')
```

File: .devops/deploy-pipelines.yml [hunk 6]
```diff
@@ -297,4 +304,255 @@ stages:
...
+              instructions: "Please approve or reject PROD green promotions v${prod_version}"
+              onTimeout: 'reject'
+
+  - stage: "Build_PROD_Green"
+    displayName: 'PROD green Build'
+    dependsOn: [ PROD_Green_WaitForApproval,Release ]
+    condition: |
+      and(
+        eq(${{parameters.UAT_PROD_DEPLOY}}, true),
+        in(dependencies.PROD_Green_WaitForApproval.result, 'Succeeded'),
```
Comment (pietro-tota) [inline .devops/deploy-pipelines.yml:448]:
```suggestion
        eq(dependencies.PROD_Green_WaitForApproval.result, 'Succeeded'),
```

File: .devops/deploy-pipelines.yml [hunk 7]
```diff
@@ -297,4 +304,255 @@ stages:
...
+              script: |
+                docker tag $(UAT_CONTAINER_NAMESPACE)/$(K8S_IMAGE_REPOSITORY_NAME):$(prod_version) $(PROD_CONTAINER_NAMESPACE)/$(K8S_IMAGE_REPOSITORY_NAME):$(prod_version)
+                docker push $(PROD_CONTAINER_NAMESPACE)/$(K8S_IMAGE_REPOSITORY_NAME):$(prod_version)
+  - stage: "Deploy_PROD_Green"
+    displayName: 'PROD green deployment'
+    dependsOn: [ Build_PROD_Green,Release ]
+    condition: |
+      and(
+        eq(${{parameters.UAT_PROD_DEPLOY}}, true),
+        in(dependencies.Build_PROD_Green.result, 'Succeeded'),
```
Comment (pietro-tota) [inline .devops/deploy-pipelines.yml:486]:
```suggestion
        eq(dependencies.Build_PROD_Green.result, 'Succeeded'),
```

**Review by pietro-tota — DISMISSED** [review]
🚀 just some trivial issues

File: .devops/azure-templates/chart-current-version.yml [hunk 8]
```diff
@@ -12,7 +12,12 @@ steps:
         CHART_FILE="helm/Chart.yaml"
         version=$(yq -r '.version' $CHART_FILE)
         appVersion=$(yq -r '.appVersion' $CHART_FILE)
+        trimmed_repo_url=$(echo $(Build.Repository.Uri) | sed -e "s/\.git//g")
+        releaseUrl=$(echo $trimmed_repo_url/releases/tag/v$appVersion-RELEASE)
+        commitUrl=$(echo $trimmed_repo_url/commit/$(Build.SourceVersion))
```
Comment (pietro-tota) [inline .devops/azure-templates/chart-current-version.yml:17]: Add commit url to UAT manual approval instruction as done here https://github.com/pagopa/pagopa-notifications-service/blob/806310ddbc95b67ddbcae80519d16489072a65f5/.devops/deploy-pipelines.yml#L243

**Pull request merged**